test/end-to-end/offer.js

```diff
@@ -633,41 +632,139 @@ describe("Offer endpoint tests", () => {
                 jobType: "SUMMER INTERNSHIP",
                 fields: ["DEVOPS", "MACHINE LEARNING", "OTHER"],
                 technologies: ["React", "CSS"],
+                isHidden: false,
                 // owner: Will be set in beforeAll,
                 location: "Testing Street, Test City, 123",
             };
             const test_offer_2 = {
                 title: "Test Offer",
-                publishDate: "2019-11-22T00:00:00.000Z",
-                publishEndDate: "2019-11-28T00:00:00.000Z",
+                publishDate: "2021-01-14T00:00:00.000Z",
+                publishEndDate: "2021-02-28T00:00:00.000Z",
                 description: "For Testing Purposes",
                 contacts: ["geral@niaefeup.pt", "229417766"],
                 jobType: "SUMMER INTERNSHIP",
                 fields: ["FRONT_END", "BACK_END"],
                 technologies: ["Flutter", "Node.js"],
+                isHidden: false,
                 // owner: Will be set in beforeAll,
                 location: "Testing Avenue, Test State, 321",
             };
+
+            const test_offer_3 = {
+                title: "Secret Offer",
+                publishDate: "2021-01-14T00:00:00.000Z",
+                publishEndDate: "2021-02-28T00:00:00.000Z",
+                description: "Secret offer that only admins and the owner can see",
+                contacts: ["geral@niaefeup.pt", "229417766"],
+                jobType: "SUMMER INTERNSHIP",
+                fields: ["FRONT_END", "BACK_END"],
+                technologies: ["Flutter", "Node.js"],
+                isHidden: true,
+                // owner: Will be set in beforeAll,
+                location: "Area 51, Secret State, 321",
+            };
+
+            let test_1, test_2, test_3;
+
+            const test_agent = agent();
+
+            const test_user_admin = {
+                email: "admin@email.com",
+                password: "password123",
+            };
+            const test_user_company = {
+                email: "company@email.com",
+                password: "password123",
+            };
+
             beforeAll(async () => {
-                await Offer.create([test_offer_1, test_offer_2]);
+                await Company.deleteMany({});
+                await Offer.deleteMany({});
+                await Account.deleteMany({});
+
+                test_company = await Company.create({
+                    name: "test company",
+                    bio: "a bio",
+                    contacts: ["a contact"]
+                });
+
+                await Account.create({
+                    email: test_user_admin.email,
+                    password: await hash(test_user_admin.password),
+                    isAdmin: true
+                });
+                await Account.create({
+                    email: test_user_company.email,
+                    password: await hash(test_user_company.password),
+                    company: test_company._id
+                });
```

Comment:
Something similar is being done a test/describe above. Is there a way we can reuse the other one (maybe by moving this logic to a common describe between both?